src/d3d9/d3d9_device.cpp

```diff
@@ -2361,11 +2361,14 @@ namespace dxvk {
           D3DPRIMITIVETYPE PrimitiveType,
           UINT             StartVertex,
           UINT             PrimitiveCount) {
-	if (unlikely(!PrimitiveCount))
-	  return S_OK;
-
     D3D9DeviceLock lock = LockDevice();
 
+    if (unlikely(m_state.vertexDecl == nullptr))
+      return D3DERR_INVALIDCALL;
+
+    if (unlikely(!PrimitiveCount))
+      return S_OK;
```

Comment:
Why move this into the lock?

Reply:
it accesses m_state so it has to be inside the lock

Reply:
Yes, what doitsujin said :frog:

Reply:
Got it.